Library/Formula/llvm.rb

```diff
@@ -94,7 +94,7 @@ class Llvm < Formula
   deprecated_option "rtti" => "with-rtti"
   deprecated_option "disable-assertions" => "without-assertions"
 
-  if MacOS.version <= :snow_leopard
+  if MacOS.version <= :snow_leopard || build.with?("python")
```

Comment:
Hmm. The logic should work like this AFAIK:

* MacOS = Snow Leopard or below? Yes => Use Homebrew Python. No => Use System Python.
* MacOS = Lion or above? Yes => Requested Python build? No => Use system. Yes => Use Homebrew's.

The second is conditional on Homebrew's Python being in your `$PATH`.

Maybe I'm wrong.